openapi/paths.go

```diff
@@ -33,6 +33,9 @@ func (p *PathItem) GetPath() string {
 }
 
 func (p *PathItem) getBasePath() string {
+	if p.parent == nil {
```

Comment:
How do you feel about maintaining the stance that any type can be the root of the tree? Hardcoding the expectation of a specific type at the root makes the project inflexible. While I can't see an immediate need for anything other than *OpenAPI to be at the root, I do prefer to keep the extensibility.

Can we define a root node as any node that has `parent == nil`?